aesara/link/utils.py

```diff
@@ -576,6 +577,23 @@ def wrapper(type, value, trace):
 register_thunk_trace_excepthook()
 
 
+@contextmanager
+def use_optimized_cheap_pass(*args, **kwargs):
+    """Temporarily replace the cheap optimization pass with a better one."""
+    from numba.core.registry import cpu_target
+
+    context = cpu_target.target_context._internal_codegen
+    old_pm = context._mpm_cheap
+    new_pm = context._module_pass_manager(
+        loop_vectorize=True, slp_vectorize=True, opt=3, cost="cheap"
+    )
+    context._mpm_cheap = new_pm
+    try:
+        yield
+    finally:
+        context._mpm_cheap = old_pm
+
+
```

Comment:
This is a general utility module for linking, so we can't put this Numba-specific code here.